### ISSUE TITLE
Report that using descriptor sets after their descriptor set layout was destroyed is invalid

### ISSUE THREAD

**Issue by alexander-a-kharlamov** [KhronosGroup/Vulkan-LoaderAndValidationLayers#2348]
This request is for the validation layer to report whenever a descriptor set is used after it's descriptor set layout has been destroyed. Spec section 2.3.1 says:

VkDescriptorSetLayout objects may be accessed by commands that operate on descriptor sets allocated using that layout, and those descriptor sets must not be updated with vkUpdateDescriptorSets after the descriptor set layout has been destroyed. Otherwise, descriptor set layouts can be destroyed any time they are not in use by an API command.

Currently the validation layer doesn't detect such cases (specifically using vkUpdateDescriptorSets) which leads to incorrect behavior / driver crashes on different platforms.

**Comment by tobine** [comment]
In addition to the validation layer check, there should be a Valid Usage case in the spec under "vkUpdateDescriptorSets()" that enforces that restriction. I'll add the check and request a spec update to that effect.

**Comment by jzulauf-lunarg** [comment]
@tobine  -- I've noted a straightfoward fix ... are you working this, or should I?

**Comment by tobine** [comment]
@jzulauf-lunarg please go ahead and add fix/check. I've submitted a spec issue to add related VUs.

**Comment by jzulauf-lunarg** [comment]
I was going overload the "must be a valid DescriptorSet" VU for now, or I can leave it as a non-VUID issue until the VUID's are added. Preference?

**Comment by tobine** [comment]
Yes, I recommend overloading that VU for now w/ TODO comment on pending VUID update when spec change lands.

**Comment by jzulauf-lunarg** [comment]
Patch available for test against original issue.
